quint/src/cliCommands.ts

```diff
@@ -66,7 +66,7 @@ interface OutputStage {
   failed?: string[]
   ignored?: string[]
   // Test names output produced by 'run'
-  status?: 'ok' | 'violation' | 'failure'
+  status?: 'ok' | 'violation' | 'failure' | 'error'
```

Comment:
What does it mean that we know allow `error` as a status in `OutputStage` and that we didn't previously?

Reply:
Also, how does this relate to the changes here? IIUC, the `runSimulator` function should only be able to return the statuses that are defined in https://github.com/informalsystems/quint/blob/d4b7b0f713bb8f1f693aa2454127f96c771b45f0/quint/src/cliCommands.ts#L148-L151

So I'm not clear on how the status in `OutputStage` comes into play.

Reply:
The `'error'` case comes from here:
https://github.com/informalsystems/quint/blob/gabriela%2Ffix-1339/quint/src/simulation.ts#L35

Previously, we had a separate error reporting for the case of `'error'`, which didn't include the status attribute (lines 534-536 before changes). Now, all errors are reported in the same manner (lines 573-579 after changes), which include the status attribute, which can now be `'error'` since it is not a special case anymore.

I guess I have to update the `SimulatorStage` type as well. It is weird to me that typescript didn't give me a type error here.

Reply:
Ah, I see why we don't need those in `SimulatorStage`: the `SimulatorStage` is only ever returned in case of success (`'ok'`). Otherwise, we call `cliErr`, which results in an `OutputStage`. In fact, we only need to have the `'ok'` option there. This typechecks:

```ts
interface SimulatorStage extends LoadedStage {
  status: 'ok'
  trace?: QuintEx[]
}
```

Anyway, I'll leave this for the future refactoring.